src/main/java/com/xebia/visualreview/PixelComparator.java

```diff
@@ -21,77 +21,149 @@
-      int[] diffData = new int[diffWidth * diffHeight];
-      int differentPixels = 0;
-
-      for (int y = 0; y < diffHeight; y++) {
-        for (int x = 0; x < diffWidth; x++) {
-          if (x >= minX || y >= minY || beforeData[y * beforeWidth + x] != afterData[y * afterWidth + x]) {
-              diffData[y * diffWidth + x] = DIFFCOLOUR;
-              differentPixels++;
-          }
+    private static int DIFFCOLOUR = (128<<24) & 0xff000000|       //a
+            (128<<16) &   0xff0000|       //r
+            (0  <<8 ) &     0xff00|       //g
+            (0  <<0 ) &       0xff;       //b
+
+    private static class Rect{
+        private int x;
+        private int y;
+        private int width;
+        private int height;
+        public boolean isValid;
+        public Rect(int x,int y,int width, int height){
+            this.x = x;
+            this.y = y;
+            this.width = width;
+            this.height = height;
+            isValid = true;
         }
-      }
+        public Rect(java.util.Map info, int maxWidth, int maxHeight){
+            isValid = true;
+            try{
+                int x = Integer.parseInt(info.get(Keyword.find("x")).toString());
+                int y = Integer.parseInt(info.get(Keyword.find("y")).toString());
+                int width = Integer.parseInt(info.get(Keyword.find("width")).toString());
+                int height = Integer.parseInt(info.get(Keyword.find("height")).toString());
 
-      BufferedImage diffImage = new BufferedImage(diffWidth, diffHeight, BufferedImage.TYPE_INT_ARGB);
-      diffImage.setRGB(0, 0, diffWidth, diffHeight, diffData, 0, diffWidth);
+                this.x = Math.min(Math.max(0,x),maxWidth);
+                this.y = Math.max(Math.min(this.y,maxHeight),y);
+                this.width = Math.min(Math.max(0,width),maxWidth-this.x);
+                this.height = Math.min(Math.max(0,height),maxHeight-this.y);
+            }catch (Exception e) {
+                System.out.println("Invalid rect "+info);
+                isValid = false;
+            }
+        }
 
-      return new DiffReport(beforeFile, afterFile, diffImage, differentPixels);
+        public void applyToImage(BufferedImage image,int rgb){
+            if (isValid){
+                for(int i = x;i<x+width;i++){
+                    for(int j = y;j<y+height;j++){
+                        image.setRGB(i,j,rgb);
+                    }
+                }
+            }
+        }
+    }
 
-    } catch (Exception e) {
-      throw new RuntimeException(e);
+    private static BufferedImage generateMask(java.util.Map maskInfo, int width, int height){
+        BufferedImage maskImage = new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
+        Rect fullRect = new Rect(0,0,width,height);
+        if (maskInfo != null) {
+            java.util.List<java.util.Map> exludeszones = (java.util.List<java.util.Map>)maskInfo.get(Keyword.find("exludeZones"));
```

Comment:
I made like this because sometimes only a crop of a image should be test, not a exclude zone so only PixelComparator will need update if mask is updated with "crop zones". 
The other main reason is because I didn't know clojure before visualReview so I made only small modifications :)